Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs

```diff
@@ -502,6 +502,12 @@ private void ProcessExecuteArguments(Context context, ObjectLiteral obj, Process
                 ProcessUnsafeOptions(context, processBuilder, unsafeOptions);
             }
 
+            // GlobalPassThroughEnvironmentVariables
+            foreach (var passThroughEnvironmentVariable in context.FrontEndHost.Configuration.Sandbox.GlobalUnsafePassthroughEnvironmentVariables)

```

Comment:
>GlobalUnsafePassthroughEnvironmentVariables [](start = 102, length = 43)

I'm not sure why it is unsafe.

Also, what about other frontend that don't use Dscript. How do they benefit from global pass-through env vars?

I thought that the modification should go to the lower layer instead of here.

Reply:
The unsafe here is intent to align with the Dscript API. 

The EnvironmentVariables for fingerprint are in Process level. So the deepest level we can go is Process. However, configuration is not available in process or process builder level, so the configuration can only access in here. I will add this code to ninja frontend as well, so we can cover both cases. Since there is no special config for process level, I will keep in in the configuration.Sandbox.

Reply:
Why don't you add it to frontend configuration? It's weird that it's property of sandbox, but it has nothing to do with our sandbox.

---
In reply to: [292672934](https://github.com/microsoft/BuildXL/pull/442#discussion_r292672934) [](ancestors = 292672934)

Reply:
moved passThroughEnvironmentVariable to frontend configuration.